src/utils.jl

```diff
@@ -70,9 +73,9 @@ _construct_without_index(avg::Average) = Average(_construct_without_index(avg.op
 function _construct_without_index(op::OperatorTerm)
     args = []
     for arg in op.arguments
-        push!(args, _construct_without_index(arg))
+        isempty(find_index(arg)) ? push!(args, arg) : push!(args, _construct_without_index(arg))
```

Comment:
Why not leave the original `push!(args, _construct_without_index(arg))` and add a fallback `_construct_without_index(x)=x` to handle non-indexed stuff?